src/Microsoft.AspNetCore.Sockets.Http/ConnectionManager.cs

```diff
@@ -94,8 +102,10 @@ public void RemoveConnection(string id)
 
         private static string MakeNewConnectionId()
         {
-            // TODO: We need to sign and encyrpt this
-            return Guid.NewGuid().ToString();
+            var id = Interlocked.Increment(ref _id);
+            // HMAC the id because we want a cryptographically random id, which GUID is not
+            var hashed = _hmacHash.ComputeHash(Encoding.UTF8.GetBytes(id.ToString()));
```

Comment:
Maybe this whole thing makes no sense and we should just use rngcryptoserviceprovider

Reply:
That's what I've been saying. If the only outcome is to generate a more-different random number, why not just do that? If the goal is to validate something, we need a different approach.